bundles/org.openhab.binding.insteon/src/main/java/org/openhab/binding/insteon/internal/device/CommandHandler.java

```diff
@@ -0,0 +1,912 @@
+                    byte value = (byte) ((cmd == OnOffType.ON) ? 0xFF : 0x00);
+                    int group = getGroup(conf);
+                    if (group == -1) {
+                        logger.warn("no group=xx specified in item {}", conf.getChannelName());
+                        return;
+                    }
+                    logger.debug("{}: sending {} broadcast to group {}", nm(), (cmd1 == 0x11) ? "ON" : "OFF",
+                            getGroup(conf));
+                    Msg m = dev.makeStandardMessage((byte) 0x0f, cmd1, value, group);
+                    dev.enqueueMessage(m, feature);
+                }
+            } catch (IOException e) {
+                logger.warn("{}: command send i/o error: ", nm(), e);
+            } catch (FieldException e) {
+                logger.warn("{}: command send message creation error ", nm(), e);
+            }
+        }
+    }
+
+    /**
+     * This Handler was supposed to set the LEDs of the 2487S, but it doesn't work.
+     * The parameters were modeled after the 2486D, it may work for that one,
+     * leaving it in for now.
+     *
+     * From the HouseLinc PLM traffic log, the following commands (in the D2 data field)
+     * of the 2486D are supported:
+     *
+     * 0x02: LED follow mask may work or not
+     * 0x03: LED OFF mask
+     * 0x04: X10 addr setting
+     * 0x05: ramp rate
+     * 0x06: on Level for button
+     * 0x07: global LED brightness (could not see any effect during testing)
+     * 0x0B: set nontoggle on/off command
+     *
+     * crucially, the 0x09 command does not work (NACK from device)
+     *
+     * @author Bernd Pfrommer - openHAB 1 insteonplm binding
+     */
+    @NonNullByDefault
+    public static class LEDOnOffCommandHandler extends CommandHandler {
+        LEDOnOffCommandHandler(DeviceFeature f) {
+            super(f);
+        }
+
+        @Override
+        public void handleCommand(InsteonChannelConfiguration conf, Command cmd, InsteonDevice dev) {
+            try {
+                int button = this.getIntParameter("button", -1);
+                if (cmd == OnOffType.ON) {
+                    Msg m = dev.makeExtendedMessage((byte) 0x1f, (byte) 0x2e, (byte) 0x00,
+                            new byte[] { (byte) button, (byte) 0x09, (byte) 0x01 });
+                    dev.enqueueMessage(m, feature);
+                    logger.debug("{}: sent msg to switch {} on", nm(), dev.getAddress());
+                } else if (cmd == OnOffType.OFF) {
+                    Msg m = dev.makeExtendedMessage((byte) 0x1f, (byte) 0x2e, (byte) 0x00,
+                            new byte[] { (byte) button, (byte) 0x09, (byte) 0x00 });
+                    dev.enqueueMessage(m, feature);
+                    logger.debug("{}: sent msg to switch {} off", nm(), dev.getAddress());
+                }
+            } catch (IOException e) {
+                logger.warn("{}: command send i/o error: ", nm(), e);
```

Comment:
This is actually here because Msg.makeMessage() throws an IOException if it can't find the message. This should never happen, except during development. Can I leave as is?

Reply:
done, created custom exception to replace IOException and left as warn.